astropy/table/pprint.py

```diff
@@ -133,8 +133,8 @@ def _auto_format_func(format_, val):
                     break
             else:
                 # None of the possible string functions passed muster.
-                raise ValueError('Unable to parse format string {0}'
-                                 .format(format_))
+                raise TypeError('unable to parse format string {0} for its '
```

Comment:
Keep as `ValueError`.   This is consistent with Python behavior.  The problem is not the *type* of the column but the *value* of the format string.